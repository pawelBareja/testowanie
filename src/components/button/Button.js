import React, { Component } from "react";

class Button extends Component {
  state = {
    label: "cos innego",
    count: 0,
  };

  handleClick = () => {
    this.setState({
      count: this.state.count + 1,
    });
  };
  render() {
    const { label, count } = this.state;
    return (
      <button onClick={this.handleClick}>
        {label} {count}
      </button>
    );
  }
}

export default Button;
