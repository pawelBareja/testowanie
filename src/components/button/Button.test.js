import React from "react";
import Button from "./Button";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Button label={"click me"} />).toJSON();
  expect(tree).toMatchSnapshot();
});
