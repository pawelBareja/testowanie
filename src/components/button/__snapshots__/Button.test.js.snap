// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  onClick={[Function]}
>
  cos innego
   
  0
</button>
`;
